src/scvi/nn/_base_components.py

```diff
@@ -61,26 +76,30 @@ def __init__(
         bias: bool = True,
         inject_covariates: bool = True,
         activation_fn: nn.Module = nn.ReLU,
+        batch_representation: str = "one-hot",
+        encode_covariates: bool = False,
     ):
         super().__init__()
         self.inject_covariates = inject_covariates
+        self.batch_representation = batch_representation
+        self.encode_covariates = encode_covariates
         layers_dim = [n_in] + (n_layers - 1) * [n_hidden] + [n_out]
 
-        if n_cat_list is not None:
-            # n_cat = 1 will be ignored
+        if n_cat_list is not None and self.batch_representation == "one-hot":
```

Comment:
I do not think you should check `batch_representation` here - if `n_cat_list` is not empty the covariates need to be accounted for here irrespective of what `batch_representation`  is - as the `batch_representation` is used in Module to determine if batch will be added to the cat list or not. So the input cat cov list will already contain batch if it is not embedded ,besides other cat covariates